### ISSUE TITLE
ttnn.mean gives incorrect result

### ISSUE THREAD

**Issue by VirdhatchaniKN** [tenstorrent/tt-metal#23876]
### Describe the bug

ttnn.batch_norm produces incorrect result when in training mode due to ttnn.mean (Refer here : [Link](https://github.com/orgs/tenstorrent/projects/91/views/7?pane=issue&itemId=113569416&issue=tenstorrent%7Ctt-metal%7C21363#:~:text=compared%20to%20bfloat16.-,Incorrect%20mean%20and%20variance%20computation%20in%20training%20mode,-On%20ensuring%20both))

### Steps to reproduce the issue

```
import torch
import pytest
import ttnn
from tests.ttnn.unit_tests.operations.eltwise.backward.utility_funcs import compare_equal


@pytest.mark.parametrize("testing_dtype", ["bfloat16", "float32"])
def test_mean(testing_dtype, device):
    torch_dtype = getattr(torch, testing_dtype)
    ttnn_dtype = getattr(ttnn, testing_dtype)
    
    torch_input = torch.ones([1, 32, 480, 640], dtype=torch_dtype) * 1.0

    golden = torch.mean(torch_input, (0, 2, 3), True)

    tt_in = ttnn.from_torch(
        torch_input,
        dtype=ttnn_dtype,
        device=device,
        layout=ttnn.TILE_LAYOUT,
        memory_config=ttnn.DRAM_MEMORY_CONFIG,
    )

    tt_result = ttnn.mean(ttnn.mean(tt_in, dim=[2, 3], keepdim=True), 0, True)
    print(tt_result, golden)

    comp_pass = compare_equal([tt_result], [golden])
    assert comp_pass

```

### Expected behavior

Expected mean result when input tensor is 1.0 is 1.0, but instead we get

- FP32 : 0.99707
- BF16 : 0.99707
